src/lib/tree/namespace/test/NamespaceMerkleTree.t.sol

```diff
@@ -252,6 +252,271 @@ contract NamespaceMerkleTreeTest is DSTest {
         assertTrue(isValid);
     }
 
+    function testVerifyLeafFiveOfEight() external {
```

Comment:
as you said we should figure out these tests are needed and document why they are needed so when they fail in the future we can better fix things

so we should create an issue imo to do so later, as I'm not sure we want to block on it due to the time constraints